libraries/chain/include/eosio/chain/finalizer_set.hpp

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include <eosio/chain/config.hpp>
+#include <eosio/chain/types.hpp>
+#include <chainbase/chainbase.hpp>
+#include <eosio/chain/authority.hpp>
+#include <eosio/chain/snapshot.hpp>
+
+#include <fc/crypto/bls_public_key.hpp>
+
+namespace eosio::chain {
+
+   struct shared_finalizer_authority {
```

Comment:
I don't understand why we have two classes `shared_finalizer_authority` and `finalizer_authority`. They seem identical to me?

Reply:
This is all copy/paste from the producer schedule and producer authority. Finalizer set was the producer schedule, and finalizer authority is from producer authority. I didn't know if the shared versions were needed, so I added them by default. If we are positive we don't need them, then I can just remove them.

Reply:
Leave for now until we decide what we are storing in chainbase. The `shared` versions are the chainbase types.